### PR TITLE
WFLY-15711 Remove references to entity bean primary key from timer se…

### PR DESCRIPTION
**Pull request by chengfang**
…rvice impl classes

https://issues.redhat.com/browse/WFLY-15711

### CODE REVIEW

File: ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/filestore/LegacyFileStore.java [hunk 1]
```diff
@@ -1,15 +1,6 @@
 package org.jboss.as.ejb3.timerservice.persistence.filestore;
```
Comment (darranl) [inline ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/filestore/LegacyFileStore.java:1]: Not related to this PR really but if this is line 1 is this class missing a copyright header?

File: ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/filestore/EjbTimerXmlParser_1_0.java [hunk 2]
```diff
@@ -155,9 +153,6 @@ private void parseTimer(XMLExtendedStreamReader reader, List<TimerImpl> timers)
                         if (loadableElements.info != null) {
                             builder.setInfo((Serializable) deserialize(loadableElements.info));
                         }
-                        if (loadableElements.primaryKey != null) {
```
Comment (darranl) [inline ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/filestore/EjbTimerXmlParser_1_0.java:158]: Are these changes going to result in previously working configurations now failing?

As this is a versioned parser is there also a schema to back it up, does there need to be a version 1.1 or are there already later versions with this removed?

File: ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/filestore/LegacyFileStore.java (hunk 1 above)
Comment (chengfang) [reply ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/filestore/LegacyFileStore.java:1]: I'll add a copyright header to this file.

File: ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/filestore/EjbTimerXmlParser_1_0.java (hunk 2 above)
Comment (chengfang) [reply ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/filestore/EjbTimerXmlParser_1_0.java:158]: The changes here are not structural changes so should still be compatible with existing config. All changes are internal implementation, and if any persisted timers contain primary key field, it will just be ignored.

**Comment by darranl** [comment]
@tadamski if you have some time could you please review this one so we can get it through.

File: ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/CalendarTimerEntity.java [hunk 3]
```diff
@@ -23,11 +23,9 @@
 
 import java.lang.reflect.Method;
 import java.util.Date;
-
 import javax.ejb.ScheduleExpression;
 
 import org.jboss.as.ejb3.timerservice.CalendarTimer;
-import org.jboss.as.ejb3.timerservice.schedule.CalendarBasedTimeout;
```
Comment (tadamski) [inline ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/CalendarTimerEntity.java:30]: @chengfang Is this change related to primary key removal?

Comment (chengfang) [reply ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/CalendarTimerEntity.java:30]: this field and related method is not used anywhere, so it was removed as a cleanup, not related to the primary key removal.

**Review by tadamski — APPROVED** [review]

**Review by pferraro — APPROVED** [review]

**Comment by pferraro** [comment]
@chengfang FYI - this has a merge conflict now.

**Comment by chengfang** [comment]
Merge conflict has been resolved.
